app/validators/efl_year_validator.rb

```diff
@@ -0,0 +1,27 @@
+class EflYearValidator < ActiveModel::EachValidator
+  YEAR_REGEX = /^\d{4}$/.freeze
+
+  def validate_each(record, attribute, value)
+    return if value.blank?
```

Comment:
Should we make this optional so we can override? it would also be easier to lift and shift one day. 

Maybe something like: 
```suggestion
   return if options[:allow_blank] && value.blank?
```